databuilder/extractor/neptune_search_data_extractor.py

```diff
@@ -0,0 +1,286 @@
+    traversal = traversal.by(
+        __.out('TAGGED_BY').has('tag_type', 'default').values('keys').dedup().fold()
+    )  # tags
+    traversal = traversal.by(
+        __.out('HAS_BADGE').values('keys').dedup().fold()
+    )  # badges
+
+    traversal = traversal.order().by(__.select('name'), Order.asc)
+
+    dashboards = traversal.toList()
+    for dashboard in dashboards:
+        dashboard['product'] = dashboard['uri'].split('_')[0]
+
+    return dashboards
+
+
+class NeptuneSearchDataExtractor(Extractor):
+    """
+    Extractor to fetch data required to support search from Neptune's graph database
+    """
+    QUERY_FUNCTION_CONFIG_KEY = 'query_function'
+    QUERY_FUNCTION_KWARGS_CONFIG_KEY = 'query_function_kwargs'
+    ENTITY_TYPE_CONFIG_KEY = 'entity_type'
+    JOB_PUBLISH_TAG_CONFIG_KEY = 'job_publish_tag'
+    MODEL_CLASS_CONFIG_KEY = 'model_class'
+
+    DEFAULT_QUERY_BY_ENTITY = {
+        'table': _table_search_query,
+        'user': _user_search_query,
+        'dashboard': _dashboard_search_query
+    }
+
+    def init(self, conf: ConfigTree) -> None:
+        self.conf = conf
+        self.entity = conf.get_string(NeptuneSearchDataExtractor.ENTITY_TYPE_CONFIG_KEY, default='table').lower()
+
+        if NeptuneSearchDataExtractor.QUERY_FUNCTION_CONFIG_KEY in conf:
+            self.query_function = conf.get(NeptuneSearchDataExtractor.QUERY_FUNCTION_CONFIG_KEY)
+        else:
+            self.query_function = NeptuneSearchDataExtractor.DEFAULT_QUERY_BY_ENTITY[self.entity]
+
+        self.job_publish_tag = conf.get_string(NeptuneSearchDataExtractor.JOB_PUBLISH_TAG_CONFIG_KEY, '')
+        self.neptune_client = NeptuneSessionClient()
+
+        neptune_client_conf = Scoped.get_scoped_conf(conf, self.neptune_client.get_scope())
+        self.neptune_client.init(neptune_client_conf)
+
+        model_class = conf.get(NeptuneSearchDataExtractor.MODEL_CLASS_CONFIG_KEY, None)
+        if model_class:
+            module_name, class_name = model_class.rsplit(".", 1)
+            mod = importlib.import_module(module_name)
+            self.model_class = getattr(mod, class_name)
+
+        self._extract_iter: Optional[Any] = None
+
+    def close(self) -> None:
+        self.neptune_client.close()
+
+    def extract(self) -> Optional[Any]:
+        if not self._extract_iter:
+            self._extract_iter = self._get_extract_iter()
+
+        try:
+            return next(self._extract_iter)
+        except StopIteration:
+            return None
+
+    def _get_extract_iter(self) -> Any:
+        if not hasattr(self, 'results'):
+            self.results = self.query_function(self.neptune_client.get_graph(), tag_filter=self.job_publish_tag)
+
+        for result in self.results:
+            if hasattr(self, 'model_class'):
+                obj = self.model_class(**result)
+                yield obj
+            else:
+                yield result
+
+    def get_scope(self) -> str:
+        return 'extractor.search_data'
```

Comment:
extractor.neptune_search_data ? the scope should be different between extractors